lib/puppet/provider/reboot/windows.rb

```diff
@@ -219,6 +220,13 @@ def pending_ccm_reboot?
     reboot
   end
 
+  def pending_domain_join?
+    path = 'SYSTEM\CurrentControlSet\Services\Netlogon\JoinDomain'
```

Comment:
I'm not sure if we also need to look at `AvoidSpnSet` as well?

The DSC resource (forked by MS employee @bcwilhite) checks that value as well in https://github.com/bcwilhite/PendingReboot/blob/aecb6e44ea3ce5d44c2c740d2c08eda79e955419/Public/Test-PendingReboot.ps1#L162-L165